lambda-notes/src/App.js

```diff
@@ -0,0 +1,140 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+import { Route } from 'react-router-dom';
+import EditNote from '../src/Components/rightContent/EditNote';
+import MyNotes from '../src/Components/rightContent/MyNotes';
+import CreateNote from '../src/Components/leftContent/CreateNote';
+import ViewNotes from '../src/Components/leftContent/ViewNotes';
+import Nav from '../src/Components/leftContent/Nav';
+import './App.css';
+
+class App extends Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			notes: [
+				{
+					_id: null,
+					title: '',
+					textBody: ''
+				}
+			],
+			error: ''
+		};
+	}
+	componentDidMount() {
+		axios
+			.get(`https://fe-notes.herokuapp.com/note/get/all`)
+			.then((res) => {
+				console.log('fetch', res.data);
+				this.setState({
+					notes: res.data
+				});
+			})
+			.catch((err) => {
+				console.log('error', err);
+			});
+	}
+	createNote = (note) => {
+		axios
+			.post('https://fe-notes.herokuapp.com/note/create', note)
+			.then((res) => {
+				console.log('responsela', res);
+				this.setState((prevState) => ({ notes: [ ...prevState.notes, note ] }));
+				this.props.history.push('/');
+			})
+			.catch((err) => console.log('error', err));
+	};
+	viewNote = (_id) => {
+		axios
+			.get(`https://fe-notes.herokuapp.com/note/get/${_id}`)
+			.then((res) => {
+				console.log('view note', res.data);
+				this.setState({ note: res.data.success });
+			})
+			.catch((err) => {
+				console.log('error', err);
+			});
+	};
+	editNote = (_id, note) => {
+		console.log('PUT note', note);
+		axios
+			.put(`https://fe-notes.herokuapp.com/note/edit/${_id}`, note)
+			.then((res) => {
+				console.log('PUT/UPDATE req response: ', res.data);
+				console.log('PUT/UPDATE ID: ', _id);
+				this.setState((theState) => {
+					let mappedArray = theState.notes.map((i) => (i._id === _id ? res.data : i));
+					return { notes: mappedArray };
+				});
+				console.log('note in PUT', note);
+				console.log('this.state.notes', this.state.notes);
+				this.props.history.push(`/viewNote/${note._id}`);
```

Comment:
It looks like there is no `this.props.history` in App.js, which may have to do with it not being a Route itself. I believe `this.props.history` only appears on components that have been rendered as Routes. You could possibly have a middle-man function in your editNote component that calls this function and holds the `this.props.history` stuff. Same for your delete.